### DISCUSSION TITLE
Locking callback function in 1.1 and 3.0 versions

### DISCUSSION THREAD

**Discussion by manoj4617** [openssl/openssl#21433] [Q&A] [answered]
In my C codebase, I was using Openssl 1.0 and now I am upgrading to Openssl 3.0, the locking callback functions which I had used in my C codebase to lock and unlock mutexes, `CRYPTO_set_locking_callback` and `CRYPTO_set_id_callback` have been deprecated.

I am confused about how to use the new threading API of Openssl 3.0 which basically says to use `OPENSSL_init_crypto` and `OPENSSL_init_ssl`, which will do all the work.

I wanted to know is there any replacement for the two functions I have mentioned for callbacks ??

**Comment by mattcaswell** [comment, marked as answer]
There is no replacement. They can simply be removed. OpenSSL 1.1.0+ uses either pthreads or windows threads as applicable automatically without the need for the callbacks.
